### PR TITLE
Prevent updates on unmounted components

### PR DESCRIPTION
**Pull request by PabloSzx**

### CODE REVIEW

File: packages/react/src/common.ts [hunk 1]
```diff
@@ -34,9 +34,11 @@ const updateReducer = (num: number): number => (num + 1) % 1_000_000;
 export function useForceUpdate({ doTimeout }: { doTimeout?: boolean } = {}) {
   const [, update] = React.useReducer(updateReducer, 0);
 
+  const isMounted = useIsMounted();
   const wasCalled = React.useRef(false);
 
   React.useEffect(() => {
+    isMounted.current = true;
```
Comment (PabloSzx) [inline packages/react/src/common.ts:41]: this is to prevent issues with react fast refresh

**Comment by vicary** [comment]
@PabloSzx Just got time to take a look at this, do we have more context about this?

I mean, if this is only about the warning, React recommends against `isMounted` checks https://github.com/facebook/react/pull/22114#issuecomment-929191048.

**Comment by vicary** [comment]
TBH I just read the [first article](https://medium.com/doctolib/react-stop-checking-if-your-component-is-mounted-3bb2568a4934) from Googling, they basically recommends aborting the fetch.

```ts
React.useEffect(() => {
  return () => {
    abortController.abort();
  };
});
```

Not sure how much it takes to add cancellable fetches all the way from core though.

When I have to choose, performance usually take precedence over noise reduction.

**Comment by vicary** [comment]
Unmounted warnings aside, I think I missed where this fixes a stopped subscription.

Does the warning also blocks subscriptions from updating other mounted components?

**Comment by PabloSzx** [comment]
this doesn't fix anything other than doing an extra check to prevent this react warning

**Comment by natew** [comment]
They are removing the error https://github.com/facebook/react/pull/22114

**Pull request closed without merging**